proposals/0000-lambda-layout.md

```diff
@@ -0,0 +1,353 @@
+      to lambda expressions but which could also have guards and multiple clauses,
+      it could be used instead of `-XLambdaCase` and would be able to match on
+      multiple patterns.
+ 2. The extension `-XMultiWayIf` essentially introduces standalone guards,
+    simplifying the use of guards that aren't at the outermost level of a
+    function declaration or case-expression. Among other things, this made it
+    easier to use guards inside of lambda expressions.
+    - If there were an expression similar to lambda expressions that could have
+      guards and wasn't required to have at
+      least one parameter, it could be used instead of `-XMultiWayIf`. This
+      includes all uses of `-XMultiWayIf`, not just those inside of lambdas
+      (see Example section).
+ 3. During the implementation of `-XLambdaCase`,
+    [some suggested](https://gitlab.haskell.org/ghc/ghc/issues/4359#note_51110)
+    allowing lambda expressions to have multiple clauses. This was not
+    implemented: The most obvious approach of turning `\` into a layout herald
+    had the disadvantage of making some common idioms invalid.
+    - This can be circumvented by introducing a new expression that isn't required
+      to be backwards compatible with existing idioms.
+
+This proposal, then, aims to overcome the shortcomings of lambda expressions
+and allows a new expression to have the same capabilities as function declarations,
+obviating the need for `-XMultiWayIf`, `-XLambdaCase`, and potentially even
+most of function declaration syntax, if a user wishes to use the new expression
+instead (see Example section). This new expression is introduced with the sequence `\ of`. It
+behaves similarly to lambda expressions, except with layout, guards and multiple clauses; for example,
+here is a definition of filter in terms of fix:
+
+```haskell
+filter = fix \of filter' _ []                 -> []
+                 filter' p (x:xs) | p x       -> x : rest
+                                  | otherwise ->     rest
+                   where rest = filter' p xs
+```
+
+The `\of` can be read as "*lambda* comprised *of* the following cases".
+
+By combining the functionality of several features into one feature in
+a way that's consistent with the rest of the language, it enables users who
+wish to use this functionality to work with a simpler and more consistent
+language.
+
+## Proposed Change Specification
+
+A new extension `-XMultiWayLambda` is implemented. Under this extension, a new
+expression is enabled, introduced by the token sequence `\ of`. The whitespace between  `\` and `of`
+is optional and may contain an arbitrary sequence of whitespace characters.
+`\ of` behaves in a way largely similar to `\`, but it is a layout herald, can have multiple
+clauses, and may contain guards (see BNF for details).
+
+Zero clauses are not permitted, as the type of the expression would be ambiguous.
```

Comment:
True, but why is it a problem? Or why is the problem different for this proposal than it is for `\case {}` or `case x of {}`? This should be discussed in the _Alternatives_ section.